vector-stores/spring-ai-elasticsearch-store/src/main/java/org/springframework/ai/vectorstore/elasticsearch/ElasticsearchVectorStore.java

```diff
@@ -208,6 +210,16 @@ public void doAdd(List<Document> documents) {
 		}
 	}
 
+	private Object getDocument(Document document, float[] embedding, String fieldName) {
+		Assert.notNull(document.getText(), "document's text must not be null");
+		if (fieldName.equals("embedding")) {
```

Comment:
I see your question on using `ElasticSearchDocument` here. Having a map could simplify things.

Reply:
Yes, I agree with that. It would definitely simplify things.
That's why I asked if we really need an ElasticSearchDocument record.